tests/linearizability/client.go

```diff
@@ -76,16 +84,45 @@ func (c *recordingClient) Put(ctx context.Context, key, value string) error {
 	callTime := time.Now()
 	resp, err := c.client.Put(ctx, key, value)
 	returnTime := time.Now()
+	if err != nil {
+		c.failed = append(c.failed, porcupine.Operation{
+			ClientId: int(c.id),
+			Input:    EtcdRequest{Op: Put, Key: key, PutData: value},
+			Call:     callTime.UnixNano(),
+			Output:   EtcdResponse{Err: err},
+		})
+		c.id = clientId.Add(1) - 1
```

Comment:
I assume this is to "create" a new client? a comment on why this is necessary would be quite useful for others to understand

Reply:
also don't know if this is confusing when you configure `config.clientCount` in a test and suddenly you have 5 more due to a failpoint.

Reply:
Adding a comment that recordingClient id might change.